### ISSUE TITLE
make the damn thing work 

### ISSUE THREAD

**Issue by 0x00002152** [0x00002152/docker-volume-ipfs#1]
add compose file
```
version: "3.8"

services:
  ipfs-docker-plugin:
    environment:
      IPFS_PATH: /ipfs-repo
      LANG: en_US.utf8
      TZ: UTC
    build:
      context: ./ipfs/
      dockerfile: Dockerfile
    hostname: ipfs-docker-plugin
    domainname: ipfs.internal
    command: bash -c "ipfs init ; /usr/local/bin/docker-volume-ipfs -mount /ipfs"
    privileged: true
    devices:
      - /dev/fuse:/dev/fuse:rwm
    volumes:
      - /run/docker/plugins:/run/docker/plugins
      - /srv/ipfs/0/ipns:/ipns
      - /srv/ipfs/0/ipfs:/ipfs
      - /srv/ipfs/0/.ipfs:/ipfs-repo
      - /var/lib/docker/volumes:/var/lib/docker-volumes
```

starts but 

```
docker logs -f 93f156ea8efc
initializing IPFS node at /ipfs-repo
Error: ipfs configuration file already exists!
Reinitializing would overwrite your keys.

IPFS > Initializing daemon...
IPFS > go-ipfs version: 0.5.0-rc4-116999a19
IPFS > Repo version: 9
IPFS > System version: amd64/linux
IPFS > Golang version: go1.13.10
IPFS > Swarm listening on /ip4/127.0.0.1/tcp/4001
IPFS > Swarm listening on /ip4/192.168.128.226/tcp/4001
IPFS > Swarm listening on /p2p-circuit
IPFS > Swarm announcing /ip4/127.0.0.1/tcp/4001
IPFS > Swarm announcing /ip4/192.168.128.226/tcp/4001
IPFS > Swarm announcing /ip4/82.102.20.175/tcp/26121
IPFS > API server listening on /ip4/127.0.0.1/tcp/5001
IPFS > WebUI: http://127.0.0.1:5001/webui
IPFS > IPFS mounted at: /ipfs
IPFS > IPNS mounted at: /ipns
IPFS > Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
IPFS > Daemon is ready
Get {aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1 map[]}
Create {aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1 map[]}
Error /ipfs/aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1 lstat /ipfs/aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: no such file or directory

```

```
ocker volume create --driver=ipfs  
Error response from daemon: create aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: VolumeDriver.Create: Error while looking for volumePath /ipfs/aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: lstat /ipfs/aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: no such file or directory
```

```
Apr 26 02:26:53 linux-zoxo dockerd[865]: time="2020-04-26T02:26:53.636538340Z" level=error msg="Handler for POST /v1.40/volumes/create returned error: create aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: VolumeDriver.Create: Error while looking for volumePath /ipfs/aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: lstat /ipfs/aa37fa488e1705cbd4a695729981d3e6952770988aad53212d1098f558d3e8c1: no such file or directory"
```

**Comment by 0x00002152** [comment]
ipfs seems to work 
```
docker exec -it e553d83f067e ipfs cat QmPXME1oRtoT627YKaDPDQ3PwA8tdP9rWuAAweLzqSwAWT/readme
Hello and Welcome to IPFS!

██╗██████╗ ███████╗███████╗
██║██╔══██╗██╔════╝██╔════╝
██║██████╔╝█████╗  ███████╗
██║██╔═══╝ ██╔══╝  ╚════██║
██║██║     ██║     ███████║
╚═╝╚═╝     ╚═╝     ╚══════╝

If you're seeing this, you have successfully installed
IPFS and are now interfacing with the ipfs merkledag!

 -------------------------------------------------------
| Warning:                                              |
|   This is alpha software. use at your own discretion! |
|   Much is missing or lacking polish. There are bugs.  |
|   Not yet secure. Read the security notes for more.   |
 -------------------------------------------------------

Check out some of the other files in this directory:

  ./about
  ./help
  ./quick-start     <-- usage examples
  ./readme          <-- this file
  ./security-notes
```

**Comment by 0x00002152** [comment]
seems reading and writing works.. 
```
dd if=/dev/urandom bs=16 count=1 | ipfs add
1+0 records in
1+0 records out
16 bytes copied, 0.00013973 s, 115 kB/s
added QmQA6NB7uHdKDKsGFq9z8CdqEtC5KKDJopbKkNnWw38a2y QmQA6NB7uHdKDKsGFq9z8CdqEtC5KKDJopbKkNnWw38a2y
 16 B / 16 B [=================================================================================================================================] 100.00%ipfs-docker-plugin:~/go/src/github.com/ipfs/go-ipfs # ipfs cat QmQA6NB7uHdKDKsGFq9z8CdqEtC5KKDJopbKkNnWw38a2y | hexdump -C
00000000  15 37 00 a3 a8 e9 e7 1f  ee 4f cd 5d a9 4f c2 c5  |.7.......O.].O..|
00000010
ipfs-docker-plugin:~/go/src/github.com/ipfs/go-ipfs # 
```

**Comment by 0x00002152** [comment]
Figured this out, mapping the directory into the container won't work, so there's essentially no need to try, but ipfs is working fine in the container as the mounts overlap the directories: 
```
ipfs-docker-plugin:/ # ipfs add /etc/group
added QmSbyCQQViYBc9oZku2dJ4jJKRJHcyF53ehNf3L2e5KaXS group
 447 B / 447 B [======================================================================================================================] 100.00%ipfs-docker-plugin:/ # cat /ipfs/QmSbyCQQViYBc9oZku2dJ4jJKRJHcyF53ehNf3L2e5KaXS
root:x:0:
shadow:x:15:
trusted:x:42:
users:x:100:
nogroup:x:65533:nobody
nobody:x:65534:
daemon:x:2:
wheel:x:499:
kmem:x:498:
lock:x:497:
tty:x:5:
utmp:x:496:
audio:x:495:
cdrom:x:494:
dialout:x:493:
disk:x:492:
input:x:491:
kvm:x:490:
lp:x:489:
render:x:488:
tape:x:487:
video:x:486:
messagebus:!:485:
bin:x:1:daemon
systemd-journal:x:484:
systemd-network:x:483:
systemd-resolve:x:482:
systemd-timesync:x:481:
systemd-coredump:x:480:
sshd:x:479:
ipfs-docker-plugin:/ # 

```

However it would seem as far as docker containers, a directory is created within IPFS, and the ID is subsequently used by the host via the docker plugin socket however the directory can't be mapped because /ipfs on the host is not the fuse mount that exists inside of the container and so it would be inaccessible. I guess the thing to do would be to change the plugin driver to instead of telling the host to access the volume via /ipfs, for the plugin daemon itself to mount -o bind /ipfs/<qmhash> </var/lib/docker/volumes> and instruct the daemon to use the path that gets created there.

**Comment by 0x00002152** [comment]
Essentially what I think it should do is: 

` cd ~ ; mkdir test ; cd test ; seq 1 15 | xargs -i -t dd if=/dev/urandom of={} bs=16 count=1 ; cd ~ ; ipfs add -r test`

```
added QmQ2AAhUBRCvTpE7FXJzj9hszR1dkumbVPnrF5qeG2Zsau test/1
added QmbQfiYWZNVju727rBFXrsz3dArYJu36X6aXF6fnL6gHgj test/10
added QmZkMLZ9q5roCNFofUPj7DY7NPFoEbcg9j9LjReYEQQQ6C test/11
added QmcnD9eEKnF8kP2wPjd9D4aJmW5uELxebgH9ojWTWVLseA test/12
added Qmb56PaLEz2dbrHYoBW3wsJyRAs8yqkNjCgjDGg3LcjjT2 test/13
added QmdEKC62u1agBnwDWevQe6x34yK9pDVWfKoennpGy69Byu test/14
added QmSAFbdpDUo4LdrqX9ddUmQjbXLkqkWj6SQLxXvHDMyPTu test/15
added QmSVtRxZ2TGaWmbPxkiQGqCJuHJnpnSrx4enTNEJYBeM6d test/2
added QmbUhvVryTMnMrrrgr1qtTfZ5wERn2J6AuourDMSYPJDLA test/3
added QmaggqJA46YwFrpW7BqqN7mKBm2dK6VdoXZi5ZQu42vmV7 test/4
added QmaH8rzsDSUn9X59nW3Evd7qNsU8yhtoovNPtju4AxotXA test/5
added QmdDt6DHLg99FfX5ZnUiCSrwnGHygHAi1tRY25np3enZnj test/6
added QmcNRFoXfJTeKdC9hax8D4devfgXbmGP9qqaaCqF9kkJDe test/7
added QmYY66VNi5b9J6uLDXAHKi2E9Ctndfo4U5qBspyh3oTfLG test/8
added Qmd7Wc2mJ7qHYPMQW9wjva5vJHagBqFhBFe2D7aD2Qw9gb test/9
added QmQvgmgoG6eHKUL9hFMegpMPqXSQjQBzWSvYP4TDDfZr9r test
```

then loop mount the directory to the docker volumes directory which is mapped into the container: 

```
ipfs-docker-plugin:~ # mkdir /var/lib/docker-volumes/test
ipfs-docker-plugin:~ # mount -o bind /ipfs/QmQvgmgoG6eHKUL9hFMegpMPqXSQjQBzWSvYP4TDDfZr9r /var/lib/docker-volumes/test
```

and this is visible from outside the container : 
```
sudo ls /var/lib/docker/volumes
21e19728dc312c5ab8fa13438cfbde0f1274f55ea9a98867b271a8be3f96a12d  a0e493918b368ddab0c9b24fb71f90999adce5957d10c51606f88a2db9be72c1
314079e278ed237dd8a8c766961134135a9b80776106844fff0bf2fe30925d32  a1ad7a07aed8d7de051861baf34ff7184af66f69c03ef1e457afe9ffec03d3d0
3af673d08fd6743ae97ddff35a2561f222d9a778bccf50dde36dec29b1f67f95  cd06c6cbca53a2699af96ef7801923046cceb48b7ef42bf644dd71f8e29decd1
3bbd83cad679f34cd5294d8290115356cf28286321bfb626f64591c157ac1e06  de1f155d5255b6d6c47ef7b634908233bb05e8872571db1e51b8383bc7e7e633
3d5bfe101d35bc868b641facee5b31b7687ab8b7668e030fb52e76d4a98b066d  e8afef83dbd0a1b4cc70f96433f3a087c1f16f2e07eb354e7a3b5cdbceea05bb
56957e2d0b9a1aa809894de242305ebcf04a50089c3903adc6a81edbb36aff35  hork
6c9801c9843def0a09e2242c770dc442582b98202b63819d567e35b3f41faa05  metadata.db
71b6fa3bdbdae411d37994029596dd050f4101d65c04ca9ca86ddafb6f199a38  test
957cce9ce067398abe79fcfcbf6c57d52030fced319cb2f490427bc676d7dddf

```

however it can't lstat the filesystem that is bind mounted: 
```
linux-zoxo:/boot/backup # ls /var/lib/docker/volumes/test/
linux-zoxo:/boot/backup # 

```
but it does work inside of the container unsurprisingly, so I guess this won't work after all 

```
ipfs-docker-plugin:~ # ls /var/lib/docker-volumes/test/
1  10  11  12  13  14  15  2  3  4  5  6  7  8  9
ipfs-docker-plugin:~ # 

```

basically this plugin sucks, the ipfs daemon in general sucks and is inconvenient and I can't really think of a better way to do this at the moment but, perhaps at some point a less shitty way to do this asides from how it was designed to work will become apparent

**Comment by 0x00002152** [comment]
one way might be to on the host to do something like: 

mount /dev/fuse -o <arguments, example_fs=/ipfs/<qmhash> but I doubt that the docker volume functionality can handle that on its own, probably expects a simple directory in /var/lib/docker/volumes to already be there, not great

**Comment by 0x00002152** [comment]
It would be nice but I guess in the meantime I will have to settle for having an install of go-ipfs on the docker host itself which sucks,

`useradd -r -g docker ipfs`

systemd target
```
[Unit]
Description=IPFS daemon for docker
After=network.target

[Service]
Type=simple
User=ipfs
WorkingDirectory=/
ExecStart=/bin/bash --login -c 'source /home/ipfs/.bashrc; /usr/local/bin/go-ipfs daemon --mount'
Restart=always
 
[Install]
WantedBy=multi-user.target

```

```
su ipfs 
cd 
/home/ipfs/go/bin/ipfs init
exit 
linux-zoxo:/home/toor/docker_ent/docker_plugins # chown -R ipfs:docker /ipfs
linux-zoxo:/home/toor/docker_ent/docker_plugins # chown -R ipfs:docker /ipns
Apr 27 19:08:43 linux-zoxo bash[1735]: Initializing daemon...
Apr 27 19:08:43 linux-zoxo bash[1735]: go-ipfs version: 0.5.0-rc4-116999a19
Apr 27 19:08:43 linux-zoxo bash[1735]: Repo version: 9
Apr 27 19:08:43 linux-zoxo bash[1735]: System version: amd64/linux
Apr 27 19:08:43 linux-zoxo bash[1735]: Golang version: go1.13.8
Apr 27 19:08:43 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=wlan0 SRC=10.0.0.24 DST=10.0.0.1 LEN=30 TOS=0x00 PREC=0x00 TTL=64 ID=64048>
Apr 27 19:08:43 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=wlan0 SRC=10.0.0.24 DST=239.255.255.250 LEN=122 TOS=0x00 PREC=0x00 TTL=1 I>
Apr 27 19:08:43 linux-zoxo kernel: IPTables-INPUT-Dropped: IN=PIA OUT= MAC= SRC=10.28.10.6 DST=224.0.0.251 LEN=72 TOS=0x00 PREC=0x00 TTL=1 ID=>
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/10.0.0.24/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/10.28.10.6/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/127.0.0.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/172.16.0.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/192.168.122.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/192.168.128.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip4/192.168.128.177/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /ip6/::1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm listening on /p2p-circuit
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/10.0.0.24/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/10.28.10.6/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/127.0.0.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/172.16.0.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/185.230.125.47/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/192.168.122.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/192.168.128.1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip4/192.168.128.177/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: Swarm announcing /ip6/::1/tcp/4001
Apr 27 19:08:46 linux-zoxo bash[1735]: API server listening on /ip4/127.0.0.1/tcp/5001
Apr 27 19:08:46 linux-zoxo bash[1735]: WebUI: http://127.0.0.1:5001/webui
Apr 27 19:08:46 linux-zoxo bash[1735]: IPFS mounted at: /ipfs
Apr 27 19:08:46 linux-zoxo bash[1735]: IPNS mounted at: /ipns
Apr 27 19:08:46 linux-zoxo bash[1735]: Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
Apr 27 19:08:46 linux-zoxo bash[1735]: Daemon is ready
lines 1295-1337/1337 (END)

```

fuck jesus christ what a fucking pain in the ass for fucks sake

**Comment by 0x00002152** [comment]
oh fuck, I just realized the thing I was actually supposed to create a service for was the god damn plugin and not go-ipfs, 

```
go get github.com/0x00002152/docker-volume-ipfs
cd ~/go/github.com/0x00002152/docker-volume-ipfs/
sed -i 's/vdemeester/0x0x00002152/g' Makefile version/version.go cmd/docker-volume-ipfs/docker-volume-ipfs.go
 sed -i 's/0x0x00002152/0x00002152/g' Makefile version/version.go cmd/docker-volume-ipfs/docker-volume-ipfs.go
make

```

modify the systemd unit 
```
mv /usr/lib/systemd/system/ipfs.service /usr/lib/systemd/system/ipfs-docker-volume-plugin.service

edit and change ExecStart: 

ExecStart=/bin/bash --login -c 'source /home/ipfs/.bashrc; /home/ipfs/go/src/github.com/0x00002152/docker-volume-ipfs/bin/docker-volume-ipfs'

sudo systemctl start ipfs-docker-volume-plugin

```

**Comment by 0x00002152** [comment]
sigh 

bashrc:
```
PATH=$HOME/go/bin:$PATH
```
also 
Apr 27 19:24:03 linux-zoxo bash[3987]: mkdir /run/docker/plugins: permission denied
`chown -R root:docker /run/docker/plugins/`
`chmod -R 770 /run/docker/plugins`

**Comment by 0x00002152** [comment]
that appears to have worked the directory permissions probably not really necessary for the run docker plugins directory thingy but whatever who the fuck cares
```
-- The start-up result is done.
Apr 27 19:27:55 linux-zoxo bash[4198]: mkdir /run/docker/plugins: permission denied
Apr 27 19:27:55 linux-zoxo bash[4198]: IPFS > Initializing daemon...
Apr 27 19:27:55 linux-zoxo bash[4198]: IPFS > go-ipfs version: 0.5.0-rc4-116999a19
Apr 27 19:27:55 linux-zoxo bash[4198]: IPFS > Repo version: 9
Apr 27 19:27:55 linux-zoxo bash[4198]: IPFS > System version: amd64/linux
Apr 27 19:27:55 linux-zoxo bash[4198]: IPFS > Golang version: go1.13.8
Apr 27 19:27:55 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=wlan0 SRC=10.0.0.24 DST=10.0.0.1 LEN=30 TOS=0x00 PREC=0x00 TTL=64 ID=2870 >
Apr 27 19:27:55 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=wlan0 SRC=10.0.0.24 DST=239.255.255.250 LEN=122 TOS=0x00 PREC=0x00 TTL=1 I>
Apr 27 19:27:55 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=br-7649bfeaa89c SRC=192.168.128.1 DST=224.0.0.22 LEN=40 TOS=0x00 PREC=0xC0>
Apr 27 19:27:55 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=virbr0 SRC=192.168.122.1 DST=224.0.0.22 LEN=40 TOS=0x00 PREC=0xC0 TTL=1 ID>
Apr 27 19:27:55 linux-zoxo kernel: IPTables-OUTPUT-Dropped: IN= OUT=wlan0 SRC=10.0.0.24 DST=224.0.0.22 LEN=40 TOS=0x00 PREC=0xC0 TTL=1 ID=0 DF>
Apr 27 19:27:55 linux-zoxo kernel: IPTables-INPUT-Dropped: IN=PIA OUT= MAC= SRC=10.28.10.6 DST=224.0.0.251 LEN=72 TOS=0x00 PREC=0x00 TTL=1 ID=>
Apr 27 19:28:05 linux-zoxo kernel: IPTables-INPUT-Dropped: IN=PIA OUT= MAC= SRC=10.28.10.6 DST=224.0.0.251 LEN=72 TOS=0x00 PREC=0x00 TTL=1 ID=>
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/10.0.0.24/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/10.28.10.6/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/127.0.0.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/172.16.0.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/192.168.122.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/192.168.128.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip4/192.168.128.177/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /ip6/::1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm listening on /p2p-circuit
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/10.0.0.24/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/10.28.10.6/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/127.0.0.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/172.16.0.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/185.230.125.47/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/192.168.122.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/192.168.128.1/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip4/192.168.128.177/tcp/4001
Apr 27 19:28:05 linux-zoxo bash[4198]: IPFS > Swarm announcing /ip6/::1/tcp/4001
Apr 27 19:28:06 linux-zoxo bash[4198]: IPFS > API server listening on /ip4/127.0.0.1/tcp/5001
Apr 27 19:28:06 linux-zoxo bash[4198]: IPFS > WebUI: http://127.0.0.1:5001/webui
Apr 27 19:28:06 linux-zoxo bash[4198]: IPFS > IPFS mounted at: /ipfs
Apr 27 19:28:06 linux-zoxo bash[4198]: IPFS > IPNS mounted at: /ipns
Apr 27 19:28:06 linux-zoxo bash[4198]: IPFS > Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
Apr 27 19:28:06 linux-zoxo bash[4198]: IPFS > Daemon is ready
lines 1355-1397/1397 (END)
```

**Comment by 0x00002152** [comment]
this probably has something to do with it, however I can't think about it right now 
```
Response:

    v1:

    {
        "Mountpoint": "/path/to/directory/on/host",
        "Err": ""
    }

    v2:

    {
        "Mountpoint": "/path/under/PropagatedMount",
        "Err": ""
    }

Respond with the path on the host (v1) or inside the plugin (v2) where the volume has been made available, and/or a string error if an error occurred.

Mountpoint is optional. However, the plugin may be queried again later if one is not provided.
```

**Comment by 0x00002152** [comment]
I can't imagine this ever worked at all, the /ipfs path has never let me access anything except a qmHash :

```
Get {e7ddc8a7f10904e96ae3783a5116e3b6919d81fa072693e17feda763f0414f6d map[]}
Create {e7ddc8a7f10904e96ae3783a5116e3b6919d81fa072693e17feda763f0414f6d map[]}
Error /ipfs/e7ddc8a7f10904e96ae3783a5116e3b6919d81fa072693e17feda763f0414f6d lstat /ipfs/e7ddc8a7f10904e96ae3783a5116e3b6919d81fa072693e17feda763f0414f6d: no such file or directory


```

**Comment by 0x00002152** [comment]
but it does work, to some extent (running as root):
```
ipfs add -r test/
added QmbhUDapY5Hxb7PT52Eiz9dPpUWa7bchmsZPw1CL4hGyDt test/1
added QmNvN7bgHD3y7U6vnC87rpLojdWDWiTpCuSJi49UvRW6Xt test/10
added QmThxnNZaB3bTaiEffTyT5Ca2JhUJNLkYHFTS4NUjgsMtm test/11
added QmStNQnuTBLfiBqhVcsBHttcYWjscojJ91cv2g3fhWWwmD test/12
added QmeDmUamEqwE4gMBrC3NtaUxhT9wVf6r2C7KfVuSACvJHS test/13
added QmdAqf9hsccsR9NLwmdbSM7WL9kipdvwurFg6gnafLvTNw test/14
added QmSwEearS7DCn4vRcP3nuweS64QnM1DkNyz6FofLTw8Cxc test/15
added QmSbLfzwfjbhVyKBAcN2szSGpnyTUY4TeVHiHgZjduxzRn test/2
added QmbXEsfXrV9rTjtoYeebVjFonbC5WK2LsRmud6ijGYSh8a test/3
added QmSonLPzsvnnJYPLtAGU8e233PCoSKBaLpWixjK6K59T7Y test/4
added QmRvggEpCFks7TCYN6KxKFcTPy9TLrvLvS1ybJu94oiXr6 test/5
added QmT94Tzota6Ue3AUfo37hV7NKxRvBYMGDNtjWr9uBq2n8D test/6
added Qmb2Dydfm98AKSLsgbZQ3iq3HDAVbkvkBtwjNYKoWBT88t test/7
added QmThJmLKPmbkYfbAyeGCPiDGE4inC9xxHvSLRxj1M8jtEw test/8
added QmUpEWitPsqqTNwTto6u23DBCW5JPsnpb2BXodDgMkn6F9 test/9
added QmVV5UwgP9mJdDbTbkGtcCR7J6mUGzheJHtTnSKbS5LkHH test

linux-zoxo:~ # docker run -it --rm -v QmVV5UwgP9mJdDbTbkGtcCR7J6mUGzheJHtTnSKbS5LkHH:/data --volume-driver=ipfs busybox ls /data
1   10  11  12  13  14  15  2   3   4   5   6   7   8   9
```
but this is far from ideal and thats literally about all it does, which is unsurprising but also completely different from how I imagined this working, the fuse filesystem doesn't even support it: 
```
linux-zoxo:~ # echo hurrrr > /ipfs/QmVV5UwgP9mJdDbTbkGtcCR7J6mUGzheJHtTnSKbS5LkHH/shutup
bash: /ipfs/QmVV5UwgP9mJdDbTbkGtcCR7J6mUGzheJHtTnSKbS5LkHH/shutup: Operation not permitted
linux-zoxo:~ # echo hurrrr > /ipfs/QmVV5UwgP9mJdDbTbkGtcCR7J6mUGzheJHtTnSKbS5LkHH/shutup^C
linux-zoxo:~ # ls /ipfs/QmVV5UwgP9mJdDbTbkGtcCR7J6mUGzheJHtTnSKbS5LkHH/
1  10  11  12  13  14  15  2  3  4  5  6  7  8  9
linux-zoxo:~ # 

```
